packages/hoppscotch-sh-admin/src/components/settings/AuthProvider.vue

```diff
@@ -86,47 +90,76 @@ const workingConfigs = useVModel(props, 'config', emit);
 const capitalize = (text: string) =>
   text.charAt(0).toUpperCase() + text.slice(1);
 
-// Masking sensitive fields
-type Field = {
+// Union type for all possible field keys
+type ProviderFieldKeys = keyof ProviderFields;
+
+type ProviderFields = {
+  [Field in keyof Config['providers'][SsoAuthProviders]['fields']]: boolean;
+} & Partial<{ tenant: boolean }>;
+
+type ProviderFieldMetadata = {
   name: string;
-  key: keyof Config['providers']['google' | 'github' | 'microsoft']['fields'];
+  key: ProviderFieldKeys;
+  applicableProviders: SsoAuthProviders[];
 };
 
-const providerConfigFields = reactive<Field[]>([
-  { name: t('configs.auth_providers.client_id'), key: 'client_id' },
-  { name: t('configs.auth_providers.client_secret'), key: 'client_secret' },
-]);
+const providerConfigFields = <ProviderFieldMetadata[]>[
+  {
+    name: t('configs.auth_providers.client_id'),
+    key: 'client_id',
+    applicableProviders: ['google', 'github', 'microsoft'],
+  },
+  {
+    name: t('configs.auth_providers.client_secret'),
+    key: 'client_secret',
+    applicableProviders: ['google', 'github', 'microsoft'],
+  },
+  {
+    name: t('configs.auth_providers.callback_url'),
+    key: 'callback_url',
+    applicableProviders: ['google', 'github', 'microsoft'],
+  },
+  {
+    name: t('configs.auth_providers.scope'),
+    key: 'scope',
+    applicableProviders: ['google', 'github', 'microsoft'],
+  },
+  {
+    name: t('configs.auth_providers.tenant'),
+    key: 'tenant',
+    applicableProviders: ['microsoft'],
+  },
+];
```

Comment:
`providerConfigFields` was mostly read-only, hence I've made it not to be `reactive`. Also, an additional `applicableProviders` field is added that decides the fields applicable for a certain provider. For instance, accounting for `Tenant` that is specific to `Microsoft`.